segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -176,7 +178,10 @@ public ChunkedSegmentStorage(int containerId, ChunkStorage chunkStorage, ChunkMe
                 chunkStorage,
                 metadataStore,
                 config,
-                executor);
+                executor,
+                System::currentTimeMillis,
```

Comment:
FYI `currentTimeMillis` is orders of magnitude slower than `nanoTime`. If all you need is a timer, consider using `AbstractTimer`; there's a `Timer` implementation using `System.nanoTime` and you can also have a `ManualTimer` (it's somewhere in the unit test package) which you can control at will.

Reply:
Also why do you expose both this and the "delayed future` below? Would it not have been simpler to create 2 constructors for this?

Reply:
System::currentTimeMillis is a method to get current time.
Delayed future is supplier of delayed execution.
These are two separate unrelated things.

Having these as suppliers allows me to precise control the sequence of steps/scenarios in unit tests.